KAFKA-10768 Make ByteBufferInputStream.read(byte[], int, int) to follow the contract

bertber opened the pull request:
I made a test for ByteBufferInputStream in the ByteBufferLogInputStreamTest.
First, I add a ByteBuffer that it's not empty to the ByteBufferInputStream, in order to verify it.
After that, I try to use ByteBufferInputStream's read function and check return value whether it's correct.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

bertber committed:
KAFKA-10768 Add a test for ByteBufferInputStream to ByteBufferLogInputStreamTest

I made a test for ByteBufferInputStream in the ByteBufferLogInputStreamTest.
First, I add a ByteBuffer that it's not empty to the ByteBufferInputStream, in order to verify it.
After that, I try to use ByteBufferInputStream's read function and check return value whether it's correct.

chia7712 commented:
@bertber Please take a look at origin PR (https://github.com/apache/kafka/pull/3752/files#diff-ae627decd5dd27d053a7a9b051860f60df6b89c8884e50df2f9ae0de6a4645e5R40). It would be better to fix the scenario of 'len==0'.

chia7712 commented on clients/src/test/java/org/apache/kafka/common/record/ByteBufferLogInputStreamTest.java:
```diff
@@ -120,4 +121,26 @@ public void iteratorRaisesOnTooLargeRecords() {
     }
 
+    @Test
+    public void testReadUnsignedIntFromInputStream() {
```
Could you move the tests to ```ByteBufferInputStreamTest```? If there is no such test class, feel free to create new one.

Commit:
Add a fixed the scenario of 'len==0' to read function.

chia7712 commented:
Is ```kafka.server.ControllerMutationQuotaTest.testPermissiveDeleteTopicsRequest``` related to this PR?

bertber commented:
> Is `kafka.server.ControllerMutationQuotaTest.testPermissiveDeleteTopicsRequest` related to this PR?

Yes, I think it is related to this PR.

Commit:
Create a ByteufferInputStreamTest class and move the
testReadUnsignedIntFromInputStream from ByteBufferLogInputStreamTest to
ByteufferInputStreamTest.

chia7712 commented on clients/src/test/java/org/apache/kafka/common/record/ByteBufferLogInputStreamTest.java:
```diff
@@ -119,5 +119,4 @@ public void iteratorRaisesOnTooLargeRecords() {
         assertNotNull(logInputStream.nextBatch());
         logInputStream.nextBatch();
     }
-
```
please revert this change.

bertber replied:
This change have been reverted.

chia7712 commented on clients/src/test/java/org/apache/kafka/common/utils/ByteBufferInputStreamTest.java:
```diff
@@ -0,0 +1,49 @@
+public class ByteBufferInputStreamTest {
+
+    @Test
+    public void testReadUnsignedIntFromInputStream() throws Exception {
```
```throws Exception ``` is useless.

bertber replied:
It have been removed.

Commit:
Revert ByteBufferLogInputStreamTest's change and remove unnecessary code
from ByteBufferInputStreamTest.

chia7712 commented on clients/src/test/java/org/apache/kafka/common/utils/ByteBufferInputStreamTest.java:
```diff
@@ -0,0 +1,49 @@
+ */
+package org.apache.kafka.common.utils;
+
+import static org.junit.Assert.assertEquals;
```
We have a pending PR offering a consistent import order (see https://github.com/apache/kafka/pull/8404#discussion_r535791013). Could you please follow the rules (although it is not merged)?

1. kafka, org.apache.kafka
1. com, net, org
1. java, javax

bertber replied:
I have put the java behind the org

chia7712 reviewed (approved):
@bertber This patch LGTM. one small comment is left. Please take a look

Commit:
reformat import order

Commit:
reformat import order again
